tools/check_documentation.py

```diff
@@ -0,0 +1,106 @@
+"""Checks for attributes and code examples in RedditBase subclasses"""
+
+import os
+import re
+import sys
+
+# This line imports from the local PRAW rather than the global installed PRAW.
+sys.path.insert(0, os.path.abspath(os.path.join(__file__, "..", "..")))
+
+from praw.models.reddit.base import RedditBase  # noqa: E402
+from praw.models.reddit.modmail import ModmailObject  # noqa: E402
+from praw.util.cache import cachedproperty  # noqa: E402
+
+
+class DocumentationChecker:
+    """Checks for code block statements and attribute tables in subclasses.
+
+    Attribute exceptions holds all exceptions. All classes and subclasses in
+    the exceptions list will be ignored.
+
+    Attribute METHOD_EXCEPTIONS holds the names of methods that will be
+    filtered out.
+    """
+
+    BASE_SEARCH_CLASS = RedditBase
+    exceptions = [
+        ModmailObject,  # is never publicly accessed
+    ]
+    HAS_CODE_BLOCK = re.compile(r".. code-block::")
+    HAS_ATTRIBUTE_TABLE = re.compile(r"Attribute[ ]+Description")
+    METHOD_EXCEPTIONS = [
+        "from_data",
+        "id_from_url",
+        "parse",
+        "sluggify",
+    ]
+    subclasses = []
+
+    @staticmethod
+    def in_list_lambda(search_list):
+        return lambda item: item not in search_list
+
+    @classmethod
+    def expand(cls, base_list, append_list):
+        base_list = list(base_list)
+        while len(base_list) > 0:
+            item = base_list.pop(0)
+            append_list.append(item)
+            base_list.extend(item.__subclasses__())
+        append_list = set(append_list)
```

Comment:
This seems a little strange because it's called `append_list but then you're putting a set into it.

Reply:
That was before the change to a set. I will rename the variable accordingly.

Reply:
On further observation, the starting variable is a list. It gets changed to a set at the last step, and it needs to be a list.